pkg/controller/etcdproxy/certs.go

```diff
@@ -110,149 +115,90 @@ func (c *EtcdProxyController) setNewAPIServerCertificates(etcdstorage *etcdstora
 	return utilerrors.NewAggregate(errs)
 }
 
-// createEtcdProxyClientCAConfigMap creates ConfigMap in controller namespace with Etcd Proxy CA bundle
-// for verifying incoming client certificates.
-func (c EtcdProxyController) createEtcdProxyClientCAConfigMap(etcdstorage *etcdstoragev1alpha1.EtcdStorage,
-	clientSingerCert []byte) error {
-	// ConfigMap in controller namespace for the etcd proxy CA certificate.
-	_, err := c.kubeclientset.CoreV1().ConfigMaps(c.config.ControllerNamespace).
-		Get(etcdProxyCAConfigMapName(etcdstorage), metav1.GetOptions{})
-	if errors.IsNotFound(err) {
-		data := map[string]string{"client-ca.crt": string(clientSingerCert)}
-		_, err = c.kubeclientset.CoreV1().ConfigMaps(c.config.ControllerNamespace).
-			Create(newConfigMap(etcdstorage, etcdProxyCAConfigMapName(etcdstorage), c.config.ControllerNamespace, data))
-		if err != nil {
-			// TODO: refactor event handling (hint: see #40).
-			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
-			return err
-		}
-		return nil
-	}
-	if err != nil {
-		// TODO: refactor event handling (hint: see #40).
-		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
-		return err
-	}
-	return nil
+// ensureServingCABundle ensures the ConfigMap contains the required Serving CA bundle.
+// The Serving CA bundle is located in the controller namespace and is used by the etcd-proxy to verify API server identity.
+func (c *EtcdProxyController) ensureServingCABundle(etcdstorage *etcdstoragev1alpha1.EtcdStorage, caBytes []byte, expiryDate time.Time) error {
+	configMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      etcdProxyCAConfigMapName(etcdstorage),
+			Namespace: c.config.ControllerNamespace,
+			// TODO: Remove this once validation is in place.
```

Comment:
validation?

Reply:
:see_no_evil: 

Will be updated, I wanted to say appending and validity verification.